modules/gcp_bigquery_big_lake_table/main.tf

```diff
@@ -12,18 +12,34 @@ resource "google_bigquery_table" "google_bigquery_table" {
   dataset_id          = var.dataset_id
   deletion_protection = var.deletion_protection
   description         = var.description
-  schema              = var.schema
   table_id            = var.name
 
+  # Terraform will detect changes to this property made by BigQuery, but we'll ignore them using the `lifecycle` block.
+  schema = var.schema
+
   external_data_configuration {
     autodetect    = var.autodetect
     connection_id = var.connection_id
     source_format = var.source_format
     source_uris   = var.source_uris
 
+    # Use the exact same schema here. This one won't be changed by BigQuery, however Terraform will still detect the changes you make on purpose to this field.
+    schema = var.schema
```

Comment:
We actually need to control the presence of this `external_data_configuration.schema` config with an input for this module because:

> Schema is disallowed for Google Cloud Bigtable, Cloud Datastore backups, Avro, Iceberg, ORC and Parquet formats. 
[Source](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_table)

Reply:
See https://github.com/honestbank/terraform-gcp-bigquery/pull/31